Pr 128 refactor

alexmiddeleer opened the pull request:
This isn't done per se, but I wanted to stop when I found the security issue (plus I need to sleep). The gist of this is mostly extracting deck CRUD operations into the deck-utils file (they'll be easier to test this way and they make the components more readable), but I fixed some lint and stuff too.

alexmiddeleer committed:
Fix import that was allowing card names with whitespace padding (#131)

* Fix import that was allowing card names with whitespace padding

* more trims

evsheffield committed:
New player guide (#132)

* partial: Add new player guide page

* partial: Styles and add banner image

* style: Spacing

* style: Fix eslint warning

* refactor: Remove unused font weights

Keep italic around just in case.

* style: Use clearer class names for styling

alexmiddeleer committed:
refactor

alexmiddeleer committed:
refactor

alexmiddeleer committed:
refactor

alexmiddeleer committed:
refactor

alexmiddeleer committed:
refactor

jtrussell commented:
Great, thanks! What was the security issue you found? Let me know when it's good for a look.

evsheffield committed:
Fix missing import on homepage

PapaCapricorn committed:
Update discord link (#136)

PapaCapricorn committed:
Contact email in welcome banner (#134)

jtrussell commented:
Also, @alexmiddeleer - you can push commits directly to this PR https://github.com/mythgard-hub/mythgard-hub/pull/128.

evsheffield committed:
Deck list style (#135)

* Cleanup styles for card name in deck list

* Fix max width of deck import text box

In Firefox this was expanding way past the width of the column.
Seems to have been less of a problem in Chrome but was still slightly
noticeable for certain window sizes..

alexmiddeleer committed:
Some fixes for deck crafting costs (#137)

alexmiddeleer committed:
Fix x cost cards (#138)

* refactor

* Revert "refactor"

This reverts commit 61ef9ff17bcb0bbe4aacc78af84be2eeb0397490.

* Show card cost as X on card page

* refactor

* Update decklist to show X for x cards

* Search for X costs cards when searching for mana costs

* refactor

* refactor

* fix tests

PapaCapricorn committed:
added Exo 2 Light for <P> tag, removed Vollkorn (#140)

evsheffield committed:
feature: Add subtype to card search (#139)

alexmiddeleer committed:
Beta banner (#141)

* Add a welcome message

* Add a little red BETA next to wordmark

* Hide message on mobile

PapaCapricorn committed:
Update variable for Articles banner (#142)

PapaCapricorn committed:
Change label on homepage to Featured Decks from Top Decks (#145)

evsheffield committed:
Update deck list sort order (#144)

* feature: Add subtype to card search

* Sort by factions, mana cost, and name in deck lists

alexmiddeleer committed:
Fix deck preview view to show the factions (#143)

PapaCapricorn committed:
Events page text formatting (#146)

* 1st attempt at Events page announcement formatting

* Add missing closing tag and reformat code

* fix link styles

alexmiddeleer committed:
sort decks on decks page

alexmiddeleer committed:
fix previous change

alexmiddeleer committed:
Fix broken deck page when no deck preview returned for deck

lily-pytel committed:
Import allow duplicate lines (#151)

* Aggregate duplicate card lines into 1

* Update the cypress test... sort of

alexmiddeleer committed:
Deck search was not showing deck creation date (#149)

alexmiddeleer committed:
Decks page paging (#150)

* Add basic paging. Not quite done

* Fix total count

* Add some separation between deck list and paging controls

* Add pagination to some decks

* Update query whitelist

lily-pytel committed:
Clear deck from deck builder (#152)

* Add confirmation messages for importing and clearing decks

* When importing a deck, don't save any of the previous cards already added to the deck

* Clear deck builder after saving

* Clear the deck in progress after going to the next page

alexmiddeleer committed:
Sometimes cards appeared in 2 columns on deckbuilder page (#153)

jtrussell committed:
Add RLS for decks (#148)

* Add RLS for decks

* Move up current user function

* Typo

* Only create decks for yourself

PapaCapricorn committed:
Add Impel to Articles page (#147)

* Add Impel to Articles page

* Let prettier format according to its wishes

alexmiddeleer committed:
Fix some tests (#155)

* Fix test that expired :facepalm:

* expect messages for travis test debugging

* Attempt to fix travis

lily-pytel committed:
Limit card quantities by rarity (#154)

* Fix path and power prop types

* Add rarity to the all cards query

* Add a constant for max cards based on rarity

* Fix import test

* Bring back those props

* Limit card quantity to max by rarity

* Limit the number of cards to import by rarity

alexmiddeleer committed:
Merge branch 'master' into pr-128-refactor

alexmiddeleer committed:
Fix some lint

alexmiddeleer committed:
Fix bug where editing deck loaded an empty deck after previously saving

alexmiddeleer committed:
RLS for card_decks
